coralproject/talk
# Username banning flow

## What does this PR do?

Lets moderators force users to update their usernames. Also removes bios to reduce moderation complexity (at Greg's suggestion.)

## How do I test this PR?

1. Flag a username.
2. Go to the moderation queue. You should see a flagged username.
3. Click the red X.
4. You should see a series of dialogs about suspending the user (new language is pending from design.)
5. Click through these dialogs. The user should receive an e-mail.
6. Log in as the newly banned user, you should see a message and a prompt to update your username.
7. Try entering an invalid username (e.g. one with a space), you should receive an error message.
8. Try entering a username already listed in the system, you should receive an error message.
9. Try entering a valid username. The page should reload and you should be logged in normally.
10. Return to the admin panel. The new username should appear as flagged.

## Note
1. The page reloading at the end of the username update flow is so we don't have to worry about updating state to change the users' status in the auth store. This use case happens pretty rarely, and I want to minimize the amount of custom action and reducer code dedicated to it.
2. As per @wyatt's recommendation that we don't handle account functions with graphql I am updating the username through a REST endpoint.
